relay-general/src/protocol/session.rs

```diff
@@ -92,6 +92,22 @@ fn is_false(val: &bool) -> bool {
     !val
 }
 
+pub enum SessionError {
+    Distinct(Uuid),
+    Aggregated(u32),
+}
+
+/// Common interface for [`SessionUpdate`] and [`SessionAggregateItem`].
+pub trait SessionLike {
+    fn started(&self) -> DateTime<Utc>;
+    fn distinct_id(&self) -> Option<&String>;
+    fn total_count(&self) -> u32;
+    fn abnormal_count(&self) -> u32;
+    fn crashed_count(&self) -> u32;
+    fn errors(&self) -> Option<SessionError>;
+    fn final_duration(&self) -> Option<(f64, SessionStatus)>;
```

Comment:
This is now an interface tailored specifically for `extract_session_metrics` (especially `errors` and `final_duration`), which has little to do with the protocol. But I still think it makes sense to have only a single function for metrics extraction, such that changes apply to both types of sessions.